pkg/codegen/nodejs/gen_program.go

```diff
@@ -376,31 +376,71 @@ type programImports struct {
 func (g *generator) collectProgramImports(program *pcl.Program) programImports {
 	importSet := codegen.NewStringSet("@pulumi/pulumi")
 	preambleHelperMethods := codegen.NewStringSet()
-	var componentImports []string
 
+	// This map tracks the package tokens by the associated import.
+	//
+	// It will have entries similar to the following:
+	//
+	// npmToPuPkgName["@pulumiverse/scaleway"] = "scaleway"
 	npmToPuPkgName := make(map[string]string)
+
+	var componentImports []string
 	seenComponentImports := map[string]bool{}
+
+	// Index known PackageReference by corresponding package name token.
+	knownPackageRefsByPkg := make(map[string]schema.PackageReference)
+	for _, pkgRef := range program.PackageReferences() {
+		// Can more than one package be bound to the same name? It seems possible in case of components.
+		knownPackageRefsByPkg[pkgRef.Name()] = pkgRef
+	}
+
+	// Collect imports for a package; optionally supply packageRef if handily available, otherwise pass nil.
+	visitPkg := func(pkg string, packageRef schema.PackageReference) {
```

Comment:
Mostly lifted as-is from Resource case so it can also be used for the Invoke case.